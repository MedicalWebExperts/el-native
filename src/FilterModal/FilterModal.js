import React, { Component } from 'react';
import { bool, string, array, func } from 'prop-types';
import { Modal, Text, View } from 'react-native';

import Theme from '../Theme';
import Switch from '../Switch/Switch';
import Button from '../Button/Button';
import { Row, Col } from '../Layout/Layout';

const theme = Theme.getTheme();
const styles = theme.filterModal;
const { colors } = theme;

class FilterModal extends Component {
  static propTypes = {
    title: string.isRequired,
    filters: array.isRequired,
    modalVisible: bool.isRequired,
    closeModal: func.isRequired,
    onApplyFilters: func.isRequired,
  };

  state = {
    selectAll: false,
    filters: this.props.filters,
  };

  onApplyFilters = () => {
    this.props.onApplyFilters(this.state.filters);
    this.props.closeModal();
  };

  updateFilter = (filter, value) => {
    const { filters } = this.state;
    filters[filters.indexOf(filter)].value = value;
    this.setState({
      filters,
    });
  };

  updateSelectAll = (value, filters) => {
    filters.forEach(filter => this.updateFilter(filter, value));
    this.setState(prevState => ({ selectAll: !prevState.selectAll }));
  };

  render() {
    const {
      title, filters, modalVisible, closeModal,
    } = this.props;
    const { selectAll } = this.state;
    return (
      <Modal animationType="slide" visible={modalVisible}>
        <Col style={styles.wrapper}>
          <Row style={{ ...styles.row, ...styles.titleWrapper }}>
            <Text style={styles.title}>{title}</Text>
            <Button
              onPress={() => {
                closeModal();
              }}
              icon="ios-close"
              textStyles={{ color: 'black', fontSize: 40 }}
              transparent
            />
          </Row>
          <View style={styles.innerWrapper}>
            <Row style={styles.row}>
              <Text style={{ ...styles.text, ...styles.selectAll }}>Select All</Text>
              <Switch
                color={colors.primary}
                value={selectAll}
                onPress={value => this.updateSelectAll(value, filters)}
                styles={styles.switch}
              />
            </Row>
            {filters.length &&
              filters.map(filter => (
                <Row key={filters.indexOf(filter)} style={styles.row}>
                  <Text style={styles.text}>{filter.name}</Text>
                  <Switch
                    color={colors.primary}
                    value={filter.value}
                    onPress={value => this.updateFilter(filter, value)}
                    styles={styles.switch}
                  />
                </Row>
              ))}
            <View style={styles.button}>
              <Button onPress={this.onApplyFilters} block text="APPLY FILTERS" />
            </View>
          </View>
        </Col>
      </Modal>
    );
  }
}

export default FilterModal;
