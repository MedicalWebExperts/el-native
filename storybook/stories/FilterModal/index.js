import React, { Component } from 'react';
import { View, TouchableHighlight, Text } from 'react-native';
import { storiesOf } from '@storybook/react-native';

import CenterView from '../CenterView';
import { FilterModal } from '../../../src';

class ModalClass extends Component {
  state = {
    modalVisible: false,
  };

  filters = [
    {
      name: 'PediatricsPediatricsPediatricsPediatricsPediatricsPediatricsPediatricsPediatrics',
      value: true,
    },
    { name: 'Plastic Surgery', value: false },
    { name: 'Gynecologist', value: false },
    { name: 'Cardiac Surgery', value: false },
    { name: 'Pediatrics', value: true },
    { name: 'Plastic Surgery', value: false },
    { name: 'Gynecologist', value: false },
    { name: 'Cardiac Surgery', value: false },
    { name: 'Pediatrics', value: true },
    { name: 'Plastic Surgery', value: false },
    { name: 'Gynecologist', value: false },
    { name: 'Cardiac Surgery', value: false },
    { name: 'Pediatrics', value: true },
    { name: 'Plastic Surgery', value: false },
    { name: 'Gynecologist', value: false },
    { name: 'Cardiac Surgery', value: false },
    { name: 'Pediatrics', value: true },
    { name: 'Plastic Surgery', value: false },
    { name: 'Gynecologist', value: false },
    { name: 'Cardiac Surgery', value: false },
  ];

  handleClose = (filters) => {
    console.log(filters);
    this.setState({ modalVisible: false });
  };

  render() {
    return (
      <View>
        <FilterModal
          modalVisible={this.state.modalVisible}
          title="Specialties"
          filters={this.filters}
          closeModal={this.handleClose}
        />
        <TouchableHighlight
          onPress={() => {
            this.setState({ modalVisible: true });
          }}
        >
          <Text>Show Modal</Text>
        </TouchableHighlight>
      </View>
    );
  }
}

storiesOf('FilterModal', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('Default', () => <ModalClass />);
