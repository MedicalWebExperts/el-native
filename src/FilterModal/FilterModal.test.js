import React from 'react';
import renderer from 'react-test-renderer';

import FilterModal from './FilterModal';

const filters = [
  { name: 'Pediatrics', value: true },
  { name: 'Plastic Surgery', value: false },
  { name: 'Gynecologist', value: false },
  { name: 'Cardiac Surgery', value: false },
];

const props = {
  modalVisible: false,
  title: 'Specialties',
  filters,
  updateFilter: () => null,
  closeModal: () => null,
};

describe('FilterModal Snapshots', () => {
  it('should render a simple FilterModal', () => {
    const component = renderer.create(<FilterModal {...props} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
