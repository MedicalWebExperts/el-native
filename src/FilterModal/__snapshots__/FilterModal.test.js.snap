// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterModal Snapshots should render a simple FilterModal 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  visible={false}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#ced4da",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "maxHeight": 60,
          "paddingHorizontal": 14,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Specialties
      </Text>
      <View
        accessibilityLabel=""
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 2,
              "borderWidth": 0,
              "elevation": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <View
            style={
              Object {
                "padding": undefined,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 40,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              ÔêÜ
            </Text>
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView
      showsVerticalScrollIndicator={false}
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 14,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#ced4da",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "maxHeight": 60,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Select All
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#2196F3"
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {},
              ]
            }
            thumbTintColor={null}
            tintColor={null}
            value={false}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#ced4da",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "maxHeight": 60,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "maxWidth": "80%",
              }
            }
          >
            Pediatrics
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#2196F3"
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {},
              ]
            }
            thumbTintColor={null}
            tintColor={null}
            value={true}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#ced4da",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "maxHeight": 60,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "maxWidth": "80%",
              }
            }
          >
            Plastic Surgery
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#2196F3"
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {},
              ]
            }
            thumbTintColor={null}
            tintColor={null}
            value={false}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#ced4da",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "maxHeight": 60,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "maxWidth": "80%",
              }
            }
          >
            Gynecologist
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#2196F3"
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {},
              ]
            }
            thumbTintColor={null}
            tintColor={null}
            value={false}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#ced4da",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "maxHeight": 60,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "maxWidth": "80%",
              }
            }
          >
            Cardiac Surgery
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#2196F3"
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {},
              ]
            }
            thumbTintColor={null}
            tintColor={null}
            value={false}
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "height": 90,
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "bottom": 25,
          "flexDirection": "row",
          "marginHorizontal": 20,
          "position": "absolute",
        }
      }
    >
      <View
        accessibilityLabel=""
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "width": "100%",
            }
          }
        >
          <Text
            disabled={false}
            style={
              Object {
                "color": "#FFF",
                "fontSize": 18,
                "fontWeight": "500",
                "textAlign": "center",
              }
            }
          >
            APPLY FILTERS
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
